src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatisticsModel.java

```diff
@@ -1,81 +1,123 @@
  * It uses a default interval but it can be set from outside to manage the interval length.
  */
 public class IntervalStatisticsModel extends AndroidViewModel {
 
-    private final List<TrackPoint> trackPoints = new ArrayList<>();
+    private static final String TAG = IntervalStatisticsModel.class.getSimpleName();
+
     private MutableLiveData<List<IntervalStatistics.Interval>> intervalsLiveData;
+    private IntervalStatistics intervalStatistics;
     private Distance distanceInterval;
+    private final Distance minGPSDistance;
+    private final ContentResolver contentResolver;
+    private ContentObserver trackPointsTableObserver;
+    private TrackPoint.Id lastTrackPointId;
+
+    private final Executor executor = Executors.newSingleThreadExecutor();
+    private HandlerThread handlerThread;
+    private Handler handler;
 
     public IntervalStatisticsModel(@NonNull Application application) {
         super(application);
+        minGPSDistance = PreferencesUtils.getRecordingDistanceInterval(PreferencesUtils.getSharedPreferences(application), application);
+        contentResolver = getApplication().getContentResolver();
     }
 
-    public MutableLiveData<List<IntervalStatistics.Interval>> getIntervalStats(boolean metricUnits, @Nullable IntervalOption interval) {
-        synchronized (trackPoints) {
-            if (intervalsLiveData == null) {
-                if (interval == null) {
-                    interval = IntervalOption.OPTION_1;
-                }
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        if (trackPointsTableObserver != null) {
+            contentResolver.unregisterContentObserver(trackPointsTableObserver);
+        }
+        if (handlerThread != null) {
+            handlerThread.getLooper().quit();
+            handlerThread = null;
+        }
+        handler = null;
+    }
 
-                intervalsLiveData = new MutableLiveData<>();
-                distanceInterval = interval.getDistance(metricUnits);
-                loadIntervalStatistics();
+    public MutableLiveData<List<IntervalStatistics.Interval>> getIntervalStats(Track.Id trackId, boolean metricUnits, @Nullable IntervalOption interval) {
+        if (intervalsLiveData == null) {
+            if (interval == null) {
+                interval = IntervalOption.OPTION_1;
             }
-            return intervalsLiveData;
+
+            intervalsLiveData = new MutableLiveData<>();
+            distanceInterval = interval.getDistance(metricUnits);
+            intervalStatistics = new IntervalStatistics(distanceInterval, minGPSDistance);
+
+            loadIntervalStatistics(trackId);
         }
-    }
 
-    private void loadIntervalStatistics() {
-        IntervalStatistics intervalStatistics = new IntervalStatistics(trackPoints, distanceInterval);
-        intervalsLiveData.postValue(intervalStatistics.getIntervalList());
+        registerTrackPointsObserver(trackId);
```

Comment:
Why move this into a separate method?
(I was somehow confused as I did not realized this method should only be called once).